app/controllers/products_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class ProductsController < ApplicationController
+  before_action :set_product, only: %i[edit update]
+
+  def new
+    @product = Product.new
+  end
+
+  def edit; end
+
+  def create
+    @product = Product.new(product_params)
+    if @product.save
+      flash[:notice] = 'Product was created successfully'
+      redirect_to '/admin'
+    else
+      render 'new'
+    end
+  end
+
+  def update
+    if @product.update(product_params)
+      flash[:notice] = 'Product was updated successfully'
+      redirect_to '/admin'
+    else
+      @product.image.purge if @product.image.id.nil?
+      render 'edit'
+    end
+  end
+
+  private
+
+  def set_product
```

Comment:
better to use get_product here

Reply:
app/controllers/products_controller.rb:34:7: C: Naming/AccessorMethodName: Do not prefix reader method names with get_.
  def get_product
      ^^^^^^^^^^^
Rubocop give an error on this naming

Reply:
try to make it just product